src/garrysmod/gamemodes/zombiesurvival/gamemode/obj_player_extend.lua

```diff
@@ -230,14 +230,21 @@ function meta:MeleeViewPunch(damage)
 end
 
 function meta:NearArsenalCrate()
+	-- Always near an arsenal crate in objective maps
+	if GAMEMODE.ObjectiveMap then
```

Comment:
This continues to support obj since we've removed the ability to buy stuff after round begins.